mantidproject/mantid
# Beam centre finder for larmor with angle

Fixes #12946 

The issue was teste by Leo Banuelos, hence needs mainly only functional testing.
# For testing

The data set to load into the sample data set is 5889.nxs
The user file is:USER_LARMORTEAM_152L_a2_6x8mm_r5889.txt

All files can be found here:\olympic\Babylon5\Scratch\Anton\Testing\12946_BeamCentreFinder
The directory also contains the Windows installer from the last Jenkins build.

Make sure that they are in your Mantid search directory.

**Test of convergence**
1. Start the ISIS SANS GUI
2. Set the instrument to LARMOR
3. Load the user file
4. Load the sample file
5. On the "Reduction Settings tab" set the Qx range to be between 0.02 and 0.1
6. On the Geometry tab, in the BeamCentre group box set -0.1 and 10 for the Angle and Y position respectively.
7. Set the Max iterations to 25
8. Hit Run and wait. 
   - Confirm that no error occurs. The run should end close to -0.26 and 3, but this has been confirmed by Leo.

**Up Down Test**
1. Follow steps 1 to 5
2. On the Geometry tab, in the BeamCentre group box set -0.26 and 10 for the Angle and Y position respectively.
3. Uncheck the Left-Right checkbox such that only Up-Down is enabled
4. Set the Max iterations to 25
5. Hit Run and wait. 
   - Confirm that no error occurs. 
   - Confirm that the angle value has not changed ( we only investigated UP/Down)

**Left Right Test**
1. Follow steps 1 to 5
2. On the Geometry tab, in the BeamCentre group box set -0.1 and 3 for the Angle and Y position respectively.
3. Uncheck the Left-Right checkbox such that only Left-Right is enabled
4. Set the Max iterations to 25
5. Hit Run and wait. 
   - Confirm that no error occurs. 
   - Confirm that the Y value has not changed ( we only investigated Left/Right direction)

**0 Iterations Test**
1. Start the ISIS SANS GUI
2. Set the instrument to LARMOR
3. Load the user file
4. Load the sample file
5. Open the sample file in the instrument window
6. Set the Max iterations to 25
7. Set Angle=0.25 and Y = 3
8 Hit Run
- Confirm that there are no errors
- Confirm in the Instrument View that the centre has moved. We should not be left of the beam stop. (Any noticable change is fine)
# For Release Notes:

See [here](http://www.mantidproject.org/index.php?title=ReleaseNotes_3_6_SANS&diff=25415&oldid=25414)
